packages/flame/lib/src/experimental/geometry/shapes/circle.dart

```diff
@@ -82,6 +82,16 @@ class Circle extends Shape {
       ..add(_center);
   }
 
+  @override
+  Vector2 nearestPoint(Vector2 point) {
+    if (_radius == 0) {
+      return _center;
+    }
+    final vectorToPoint = point - _center;
+    final distance = vectorToPoint.length;
+    return _center + vectorToPoint * (_radius / distance);
```

Comment:
This creates a lot of new `Vector2` objects, you should be able to create just one and use `setFrom` and then `sub` etc instead of the operators.

Reply:
Could `length2` be used here instead? 🤔